pkgs/tools/misc/lesspipe/default.nix

```
@@ -28,9 +47,62 @@ stdenv.mkDerivation rec {
   installFlags = [ "DESTDIR=$(out)" ];
 
   postInstall = ''
-    for f in lesspipe.sh lesscomplete; do
-      wrapProgram "$out/bin/$f" --prefix-each PATH : "${lib.makeBinPath [ file gnused procps ]}"
-    done
+    # resholve doesn't see strings in an array definition
+    substituteInPlace $out/bin/lesspipe.sh --replace 'nodash strings' "nodash ${binutils-unwrapped}/bin/strings"
+
+    ${resholve.phraseSolution "lesspipe.sh" {
+      scripts = [ "bin/lesspipe.sh" ];
+      interpreter = "${bash}/bin/bash";
+      inputs = [
+        coreutils
+        file
+        gnugrep
+        gnused
+        gnutar
+        iconv
+        procps
+        ncurses
+        locale
+      ];
```

```suggestion
      ] ++ lib.optional stdenv.isDarwin [
        locale
      ];
```

I tested the new commit on my system and got the following diff

```
locale-glibc: ∅ → 2.38-27
```

which means locale is otherwise not used on my system. I think this is something darwin specific.

If there is still something missing after that, we should ask in the darwin channel for help.

---

https://github.com/wofr06/lesspipe/blob/lesspipe/lesspipe.sh#L799 is the usage site. Do you have lesspipe disabled on your system? I can't see how it would would without replacement.

---

I can see `locale` being dragged in for other reasons on non-darwin configurations.

---

Curious:
```bash
$ bash

$ a=$(false) || a=utf-8

$ echo $? && echo $a
0
utf-8

$ a=$(commandnotfound) || a=utf-8
commandnotfound: command not found

$ echo $? && echo $a
0
utf-8
```

---

For non Darwin it is probably part of glibc.

Yes I have it enabled but as seen before, it only fails on Darwin without it.